asserts/system_user_test.go

```diff
@@ -267,3 +271,45 @@ func (s *systemUserSuite) TestSuggestedFormat(c *C) {
 	c.Assert(err, IsNil)
 	c.Check(fmtnum, Equals, 1)
 }
+
+func (s *systemUserSuite) TestNoUserValidFor(c *C) {
+	s.systemUserStr = strings.Replace(s.systemUserStr, s.userValidForLine, "", 1)
+	a, err := asserts.Decode([]byte(s.systemUserStr))
+	c.Assert(err, IsNil)
+	c.Check(a.Type(), Equals, asserts.SystemUserType)
+	su := a.(*asserts.SystemUser)
+	c.Check(su.Expiration(time.Now()), Equals, time.Time{})
+}
+
+func (s *systemUserSuite) TestUserValidForExpiration(c *C) {
+	a, err := asserts.Decode([]byte(s.systemUserStr))
+	c.Assert(err, IsNil)
+	c.Check(a.Type(), Equals, asserts.SystemUserType)
+	su := a.(*asserts.SystemUser)
+	c.Check(su.Expiration(time.Now()), Equals, s.until)
+}
+
+func (s *systemUserSuite) TestUserValidForDuration(c *C) {
+	s.systemUserStr = strings.Replace(s.systemUserStr, s.userValidForLine, "user-valid-for: 24h\n", 1)
+	a, err := asserts.Decode([]byte(s.systemUserStr))
+	c.Assert(err, IsNil)
+	c.Check(a.Type(), Equals, asserts.SystemUserType)
+	su := a.(*asserts.SystemUser)
+	now := time.Now()
+	c.Check(su.Expiration(now), Equals, now.Add(24*time.Hour))
+}
+
+func (s *systemUserSuite) TestUserValidForInvalidValue(c *C) {
```

Comment:
this should be folded into TestDecodeInvalid and also it seems there are more error cases to cover

Reply:
Fixed, really nice suggestion